src/stock-adjustment-mobile/program-select.jsx

```diff
@@ -45,32 +45,55 @@ const ProgramSelect = ({ offlineService, stockReasonsFactory,
-                        dispatch(setReasons(mappedReasons));
-                        return mappedReasons
-                    }).then(mappedReasons => {
-                        goToProductPage(programId, programObject, program);
-                    });
-                }
-                else {
-                    stockReasonsFactory.getReasons(program.id, facility.type.id, reasonType).then(reasons => {
-                        const mappedReasons = _.map(reasons.filter(reason => reason.name.contains('Transfer ')), reason => ({ name: reason.name, value: reason }));
-                        dispatch(setReasons(mappedReasons));
-                        return mappedReasons
-                    }).then(mappedReasons => {
-                        goToProductPage(programId, programObject, program);
-                    });
-                }   
+            if (adjustmentType === ADJUSTMENT || adjustmentType === ISSUE) {
+                existingStockOrderableGroupsFactory.getGroupsWithNotZeroSoh(stateParams, program, facility).then(groups => {
+                    const productOptions = _.map(groups, group => ({ name: group[0].orderable.fullProductName, value: group }));
+                    dispatch(setProductOptions(productOptions));
+                    return productOptions;
+                }).then(productOptions => {
+                    afterSelectingProducts(programId, programObject, program); 
+                });
+            }
+            else {
+                orderableGroupService.findAvailableProductsAndCreateOrderableGroups(program.id, facility.id, true).then(groups => {
+                    const productOptions = _.map(groups, group => ({ name: group[0].orderable.fullProductName, value: group }));
+                    dispatch(setProductOptions(productOptions));
+                    return productOptions;
+                }).then(productOptions => {
+                    afterSelectingProducts(programId, programObject, program); 
+                });
+            }
+        }
+    };
+
+    const afterSelectingProducts = (programId, programObject, program) => {
+        const reasonType = adjustmentType === ADJUSTMENT ? null : (adjustmentType === ISSUE ? DEBIT : CREDIT ); 
+        if (!reasonType) {
+            stockReasonsFactory.getAdjustmentReasons(program.id, facility.type.id).then(reasons => {
+                const mappedReasons = _.map(reasons, reason => ({ name: reason.name, value: reason }));
+                dispatch(setReasons(mappedReasons));
+                return mappedReasons
+            }).then(mappedReasons => {
+                goToProductPage(programId, programObject, program);
             });
         }
+        else {
+            stockReasonsFactory.getReasons(program.id, facility.type.id, reasonType).then(reasons => {
+                const mappedReasons = _.map(reasons.filter(reason => reason.name.contains('Transfer ')), reason => ({ name: reason.name, value: reason }));
```

Comment:
Please change this filtering as we discussed. We cannot hardcode it.

Reply:
ok,